demo/SafetyState.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  SafetyState.swift
+//  demo
+//
+//  Created by Maksim on 11/2/18.
+//  Copyright © 2018 Mapbox. All rights reserved.
+//
+
+import Foundation
+import MapboxVision
+
+enum SafetyState {
+    
+    enum Object {
+        case car(CGRect)
+        case person(CGRect)
+    }
+    
+    enum Collision {
+        case warning(Object)
+        case critical(Object)
+    }
+    
+    case none
+    case distance(frame: CGRect, distance: Double)
+    case collisions([Collision])
+    
+    var canvasSize: CGSize {
+        return VisionManager.shared.frameSize
+    }
+    
+    init(_ worldDescription: WorldDescription) {
+        
+        let collisions = worldDescription.collisionObjects.compactMap { collision -> Collision? in
+            let bbox = collision.object.detection.boundingBox
+            let type = collision.object.detection.objectType
+            let state = collision.state
+            
+            switch (type, state) {
+            case (.lights, _), (.sign, _), (.bicycle, _), (_, .notTriggered):
+                return nil
+            case (.car, .warning):
+                return .warning(.car(bbox))
+            case (.car, .critical):
+                return .critical(.car(bbox))
+            case (.person, .warning):
+                return .warning(.person(bbox))
+            case (.person, .critical):
+                return .critical(.person(bbox))
+            }
+        }
+        
+        if collisions.count > 0 {
+            self = .collisions(collisions)
+        } else if let car = worldDescription.getForwardCar() {
+            self = .distance(frame: car.detection.boundingBox, distance: car.distance)
+        } else {
+            self = .none
+        }
+    }
+}
+
+extension SafetyState: Equatable {
```

Comment:
Since all the associated values are `Equatable` we may rely on automatic synthesis of `==`. It should be enough to specify conformance to `Equatable` at declaration.

Reply:
It even compiles for me without `Equatable` conformance. Is it used anywhere?